src/main/java/com/denizenscript/depenizen/bukkit/properties/towny/TownyLocationProperties.java

```diff
@@ -1,115 +1,76 @@
 import com.palmergames.bukkit.towny.exceptions.NotRegisteredException;
 import com.palmergames.bukkit.towny.object.TownBlock;
 import com.palmergames.bukkit.towny.TownyUniverse;
 import com.denizenscript.denizen.objects.LocationTag;
-import com.denizenscript.denizencore.utilities.debugging.Debug;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.tags.Attribute;
 
 import java.util.UUID;
 
-public class TownyLocationProperties implements Property {
-
-    @Override
-    public String getPropertyString() {
-        return null;
-    }
-
-    @Override
-    public String getPropertyId() {
-        return "TownyLocation";
-    }
-
-    @Override
-    public void adjust(Mechanism mechanism) {
-        // None
-    }
-
-    public static boolean describes(ObjectTag object) {
-        return object instanceof LocationTag;
-    }
-
-    public static TownyLocationProperties getFrom(ObjectTag object) {
-        if (!describes(object)) {
-            return null;
-        }
-        else {
-            return new TownyLocationProperties((LocationTag) object);
-        }
-    }
-
-    public static final String[] handledTags = new String[] {
-            "has_town", "town", "is_wilderness", "towny"
-    };
-
-    public static final String[] handledMechs = new String[] {
-    }; // None
-
-    private TownyLocationProperties(LocationTag location) {
-        this.location = location;
-    }
-
-    public LocationTag location;
-
-    @Override
-    public ObjectTag getObjectAttribute(Attribute attribute) {
+public class TownyLocationProperties {
+    public static void register() {
+        // <--[tag]
+        // @attribute <LocationTag.towny_allows_pvp>
+        // @returns ElementTag(Boolean)
+        // @plugin Depenizen, Towny
+        // @description
+        // Returns whether Towny would allow PVP here.
+        // -->
+        LocationTag.tagProcessor.registerTag(ElementTag.class, "towny_allows_pvp", (attribute, location) -> {
+            return new ElementTag(TownyAPI.getInstance().isPVP(location));
+        });
 
-        if (attribute.startsWith("towny")) {
-            attribute = attribute.fulfill(1);
+        LocationTag.tagProcessor.registerTag(PlayerTag.class, "towny", (attribute, location) -> {
             try {
-                TownBlock block = TownyAPI.getInstance().getTownBlock(location);
-                if (block == null) {
-                    return null;
-                }
-
                 // <--[tag]
                 // @attribute <LocationTag.towny.resident>
                 // @returns PlayerTag
                 // @plugin Depenizen, Towny
                 // @description
                 // Returns the resident of a Towny plot at the location, if any.
                 // -->
-                if (attribute.startsWith("resident")) {
+                if (attribute.startsWith("resident", 2)) {
```

Comment:
This legacy subtag needs an `attribute.fulfill(1)` - without it, the tag won't work. Have you tested the things changed in this PR?

Reply:
also should probably be deprecated and a replacement named `towny_resident` added

Reply:
> This legacy subtag needs an `attribute.fulfill(1)` - without it, the tag won't work. Have you tested the things changed in this PR?

As stated above; was unsure how this works in the rewritten system hence request for review of this. Will be done now that I do.